src/pages/workspace/rules/RulesCustomNamePage.tsx

```diff
@@ -43,17 +46,52 @@ function RulesCustomNamePage({route}: RulesCustomNamePageProps) {
 
     const customNameDefaultValue = Str.htmlDecode(policy?.fieldList?.[CONST.POLICY.FIELDS.FIELD_LIST_TITLE].defaultValue ?? '');
 
-    const validateCustomName = ({customName}: FormOnyxValues<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM>) => {
-        const errors: FormInputErrors<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM> = {};
-        if (!customName) {
-            errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.fieldRequired');
-        } else if (customName.length > CONST.WORKSPACE_NAME_CHARACTER_LIMIT) {
-            errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.characterLimitExceedCounter', {
-                length: customName.length,
-                limit: CONST.WORKSPACE_NAME_CHARACTER_LIMIT,
-            });
+    const validateCustomName = useCallback(
+        ({customName}: FormOnyxValues<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM> = {};
+            if (!customName) {
+                errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.fieldRequired');
+            } else if (customName.length > CONST.WORKSPACE_NAME_CHARACTER_LIMIT) {
+                errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.characterLimitExceedCounter', {
+                    length: customName.length,
+                    limit: CONST.WORKSPACE_NAME_CHARACTER_LIMIT,
+                });
+            }
+            return errors;
+        },
+        [translate],
+    );
+
+    const clearFieldListError = () => {
+        clearPolicyErrorField(policyID, 'fieldList');
+    };
+
+    // Get the actual error from the nested structure
+    const fieldListErrors = policy?.errorFields?.fieldList as Record<string, unknown> | undefined;
+    const textTitleErrors = fieldListErrors?.[CONST.POLICY.FIELDS.FIELD_LIST_TITLE] as Record<string, unknown> | undefined;
+    const defaultValueErrors = textTitleErrors?.defaultValue as Record<string, string> | undefined;
+
+    // Get pending action for loading state
+    const isLoading = !!policy?.fieldList?.[CONST.POLICY.FIELDS.FIELD_LIST_TITLE]?.pendingFields?.defaultValue;
+    const prevIsLoading = usePrevious(isLoading);
+
+    // Clear errors when modal is dismissed
+    useBeforeRemove(() => {
+        clearFieldListError();
+    });
+
+    useEffect(() => {
+        if (!prevIsLoading || isLoading) {
+            return;
+        }
+
+        if (!defaultValueErrors) {
+            Navigation.goBack();
         }
-        return errors;
+    }, [isLoading, prevIsLoading, defaultValueErrors]);
```

Comment:
Instead of doing conditional navigation like this after the backend responds, I think we should follow more typical patterns in the application where we can optimistically navigate the user away and assume that the backend operation will succeed. And then we use the red brick road errors to lead them back to where the error occurred when there is one. That means that we show a little red dot on every navigation item from the homepage of the app until the page where the error should be shown.